services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,271 @@
+import codecs
+from datetime import datetime
+import git
+import json
+import logging
+import os
+import shutil
+import sys
+import uuid
+
+from requests.exceptions import HTTPError
+from flask import Flask, Response
+from flask.ext.cors import CORS
+
+from dictionary_parser import Parser as DictionaryParser
+from report_parser import Parser, logger
+from timer import ResourceTimer
+
+# Token for GitHub authorization. Set on startup.
+GITHUB_AUTH_TOKEN = None
+
+DIR_NAME = 'dqa_repo'
+REMOTE_URL = 'github.com/PEDSnet/Data-Quality.git'
+
+
+class DQAResource():
+    def __init__(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+        self.dictionary = {}
+
+        self.dir_name = DIR_NAME
+        self.remote_url = 'https://' + GITHUB_AUTH_TOKEN + '@' + REMOTE_URL
+
+    def process_dqa(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+
+        dir_name = './dqa_repo/SecondaryReports/'
+        hospitals = filter(
```

Comment:
This expression is pretty confusing to decipher. Moving this into the `for` loop below would be more clear:

```python
for path in os.listdir(dir_name):
    # Full path relative to directory.
    path = os.path.join(dir_name, path)
    
    if path == 'Ranking' or !os.path.isdir(path):
        continue

    for etl_version in sorted(os.lisdir(path), reverse=True):
        #...
```